docs/mosaicking.rst

```diff
@@ -0,0 +1,310 @@
+    CTYPE1  = 'GLON-TAN'           / galactic longitude, gnomonic projection
+    CTYPE2  = 'GLAT-TAN'           / galactic latitude, gnomonic projection
+    CRVAL1  =      15.116960053834 / [deg] Coordinate value at reference point
+    CRVAL2  =    -0.72166403860488 / [deg] Coordinate value at reference point
+    LONPOLE =                180.0 / [deg] Native longitude of celestial pole
+    LATPOLE =    -0.72166403860488 / [deg] Native latitude of celestial pole
+    >>> shape_out
+    (2623, 2424)
+
+Orientation
+-----------
+
+As mentioned above, by default the image will be lined up so that North is up,
+but this may not always be optimal because if the mosaic is rotated compared to
+North, there may be a lot of empty space in the final mosaic. The ``auto_rotate``
+option can therefore be used to find the optimal rotation for the WCS that minimizes
+the area of the final mosaic file:
+
+.. doctest-requires:: shapely
+
+    >>> wcs_out, shape_out = find_optimal_celestial_wcs(m17_hdus,
+    ...                                                 frame=Galactic(),
+    ...                                                 auto_rotate=True)
+
+Note that this requires `Shapely <...>`_ 1.6 or later to be installed. We can now
+look at the final WCS and shape:
+
+.. doctest-requires:: shapely
+
+    >>> wcs_out.to_header()  # doctest: +FLOAT_CMP
+    WCSAXES =                    2 / Number of coordinate axes
+    CRPIX1  =      1102.3949574309 / Pixel coordinate of reference point
+    CRPIX2  =      900.46211361965 / Pixel coordinate of reference point
+    PC1_1   =     0.88188439264557 / Coordinate transformation matrix element
+    PC1_2   =     0.47146571244169 / Coordinate transformation matrix element
+    PC2_1   =    -0.47146571244169 / Coordinate transformation matrix element
+    PC2_2   =     0.88188439264557 / Coordinate transformation matrix element
+    CDELT1  =     -0.0002777777845 / [deg] Coordinate increment at reference point
+    CDELT2  =      0.0002777777845 / [deg] Coordinate increment at reference point
+    CUNIT1  = 'deg'                / Units of coordinate increment and value
+    CUNIT2  = 'deg'                / Units of coordinate increment and value
+    CTYPE1  = 'GLON-TAN'           / galactic longitude, gnomonic projection
+    CTYPE2  = 'GLAT-TAN'           / galactic latitude, gnomonic projection
+    CRVAL1  =      15.116960053834 / [deg] Coordinate value at reference point
+    CRVAL2  =    -0.72166403860488 / [deg] Coordinate value at reference point
+    LONPOLE =                180.0 / [deg] Native longitude of celestial pole
+    LATPOLE =    -0.72166403860488 / [deg] Native latitude of celestial pole
+    >>> shape_out
+    (1800, 2201)
+
+As expected, the optimal shape is smaller than was returned previously.
+
+Resolution
+----------
+
+By default, the final mosaic will have the resolution of the highest resolution
+input image, but this can be overriden using the ``resolution=`` keyword
```

Comment:
maybe describe what `resolution` means more specifically.  I assume it is the pixel scale along the pixel axes?  (i.e., not diagonal to the pixel, or anything weird like that)

Actually, if it _is_ pixel scale, I prefer using that term over `resolution`, since we typically use `resolution` to refer to the telescope-optics-produced PSF scale rather than the pixel scale.

Reply:
Ok I've tried to clarify things here - if we want to change the name of the actual keyword argument, we should do that in a separate PR with a deprecation phase, because that function (find_optimal...) is already in a released version of reproject (I'm just adding the docs for that here)